From 48f132159293efd30a7250b5c3b833953ebbd87e Mon Sep 17 00:00:00 2001
From: Andi Kleen <andi@firstfloor.org>
Date: Sun, 28 Mar 2021 10:21:15 -0700
Subject: Kbuild, lto: Disable LTO for bounds/asm-offset

Disable LTO when generating the bounds/asm-offset.s files which
are scanned for C constants. With a LTO build the file would contain
the gcc IR in assembler form, which breaks the scanning scripts.

Signed-off-by: Andi Kleen <andi@firstfloor.org>
---
 Kbuild | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Kbuild b/Kbuild
index fa441b98c9f6e..a24c8087ace66 100644
--- a/Kbuild
+++ b/Kbuild
@@ -10,6 +10,8 @@ bounds-file := include/generated/bounds.h
 always-y := $(bounds-file)
 targets := kernel/bounds.s
 
+kernel/bounds.s: KBUILD_CFLAGS += $(DISABLE_LTO)
+
 $(bounds-file): kernel/bounds.s FORCE
 	$(call filechk,offsets,__LINUX_BOUNDS_H__)
 
@@ -32,6 +34,7 @@ always-y += $(offsets-file)
 targets += arch/$(SRCARCH)/kernel/asm-offsets.s
 
 arch/$(SRCARCH)/kernel/asm-offsets.s: $(timeconst-file) $(bounds-file)
+arch/$(SRCARCH)/kernel/asm-offsets.s: KBUILD_CFLAGS += $(DISABLE_LTO)
 
 $(offsets-file): arch/$(SRCARCH)/kernel/asm-offsets.s FORCE
 	$(call filechk,offsets,__ASM_OFFSETS_H__)
-- 
cgit 1.2.3-1.el7

